### PR TITLE
create task, tasklist and command class with add, delete, list, mark, unmark command

### LINKED ISSUE #6: What will all the commands be?
### **The index will be extracted in Parser class and returned as a parameter. There is no need to care about index extraction in the individual classes**  

Hi all, I would like to recommend the command names for the following:
**(i) Patients (State: 0)** Chee Fong
1. Add Patients
 AddPatientCommand() _via add keyword_
 2. List Patients
  ListPatientCommand() _via list keyword_
  3. Delete Patients
  DelPatientsCommand() _via delete keyword_
  4 Select Patients
  SelectPatientCommand() _via select keyword_
  
  **(ii) Tasks (State: 1)** Thu
  1. Add Tasks
   AddTaskCommand() _via add keyword_
   2. List Tasks
   ListTaskCommand() _via list keyword_
  3. Delete Tasks
  DelTaskCommand() _via deletekeyword_
   4. Mark Tasks
  MarkTaskCommand() _via mark_
     4. Unmark Tasks
  UnmarkTaskCommand() _via unmark_
 

After the functions are defined, we will then integrate these completed parts into the main function.

### LINKED ISSUE #15: Creation of Task, TaskList class and related commands
**Description:**
Created Task class and TaskList, along with add, delete, list, mark, unmark command for the taskList.

**Features:**
Task class: Include task description, located under the data/task package.
TaskList class: has an ArrayList of class and commands to manipulate the list.

IllegalValueException: under exception package, for illegal value i.e index of not found task.

AddTaskCommand: Command to add a new task.
DeleteTaskCommand: Command to delete a task.
ListTaskCommand: Command to list all tasks.
MarkTaskCommand, UnmarkTaskCommand: Command to mark a task as done or not.

**Tasks:**

Task Class Creation:

Attributes include description, boolean isDone.
Methods to get and set these attributes.

TaskList Class:

Attributes include an ArrayList of Tasks.
Constructor with empty list or load an existing list.
Methods to add, delete, get, mark and unmark tasks.
Exceptions to check if task already existed (matching description) and task not found (index inputed out of bounds).

Command Setup:

AddTaskCommand: Command to add a new patient with input details (either an object task or take a string exception).
DeleteTaskCommand: Command to remove a patient by index.
ListTaskCommand: Command to list all tasks.
MarkTaskCommand/UnmarkTaskCommand: Command to mark a task as done or undone.

### PR DESCRIPTION
**Pull request by Metanyu**
**Description:**
Created Task class and TaskList, along with add, delete, list, mark, unmark command for the taskList.

**Features:**
Task class: Include task description, located under the data/task package.
TaskList class: has an ArrayList of class and commands to manipulate the list.

IllegalValueException: under exception package, for illegal value i.e index of not found task.

AddTaskCommand: Command to add a new task.
DeleteTaskCommand: Command to delete a task.
ListTaskCommand: Command to list all tasks.
MarkTaskCommand, UnmarkTaskCommand: Command to mark a task as done or not.

**Tasks:**
Task Class Creation:

Attributes include description, boolean isDone.
Methods to get and set these attributes.

TaskList Class: 
Attributes include an ArrayList of Tasks.
Constructor with empty list or load an existing list.
Methods to add, delete, get, mark and unmark tasks.
Exceptions to check if task already existed (matching description) and task not found (index inputed out of bounds).

Command Setup:

AddTaskCommand: Command to add a new patient with input details (either an object task or take a string exception).
DeleteTaskCommand: Command to remove a patient by index.
ListTaskCommand: Command to list all tasks.
MarkTaskCommand/UnmarkTaskCommand: Command to mark a task as done or undone.

### CODE REVIEW

**Review by lowjunchen — APPROVED** [review]

File: src/test/java/seedu/duke/commands/TaskCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        command.execute();
+
+        ListTaskCommand listTaskCommand = new ListTaskCommand();
+        listTaskCommand.setData(tasks);
+        CommandResult commandResult = listTaskCommand.execute();
+        assertEquals("Here are the tasks in your list!\n1. [ ] Read book\n", commandResult.getFeedbackToUser());
+    }
+
+    @Test
+    void testDuplicateTask(){
```
Comment (yuanners99) [inline src/test/java/seedu/duke/commands/TaskCommandTest.java:56]: If the program disallows duplication, consider edge cases such as entries with the same spelling but different letter casing.

**Review by yuanners99 — COMMENTED** [review]
Overall, the code is clean and well-structured. One suggestion would be to add more comments to enhance readability and include additional test cases to ensure comprehensive coverage of the functionality.
